samples/net/sockets/echo/src/socket_echo.c

```diff
@@ -24,24 +24,31 @@
 int main(void)
 {
 	int serv;
-	struct sockaddr_in bind_addr;
+	struct sockaddr_in6 bind_addr = {
+		.sin6_family = AF_INET6,
+		.sin6_addr = IN6ADDR_ANY_INIT,
+		.sin6_port = htons(4242),
+	};
 
-	serv = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
+	serv = socket(AF_INET6, SOCK_STREAM, IPPROTO_TCP);
 
-	bind_addr.sin_family = AF_INET;
-	bind_addr.sin_addr.s_addr = htonl(INADDR_ANY);
-	bind_addr.sin_port = htons(4242);
-	bind(serv, (struct sockaddr *)&bind_addr, sizeof(bind_addr));
+//	bind_addr.sin6_family = AF_INET6;
```

Comment:
Oops, sorry, that's a stray local change. I didn't mean to change samples/net/sockets/echo (since there's no suitable way to make it work with both IPv4 and IPv6), only the echo_async sample. Will revert this change.